pkg/updater/service/datadog_agent_test.go

```diff
@@ -0,0 +1,70 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+// Package service provides a way to interact with os services
+package service
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+)
+
+func TestInstallSignature(t *testing.T) {
+	installSigFile = filepath.Join(t.TempDir(), "/install.json")
+	require.Nil(t, writeInstallSignature())
+
+	content, err := os.ReadFile(installSigFile)
+	if err != nil {
+		require.Nil(t, err)
+	}
+	var installSignature map[string]string
+	err = json.Unmarshal(content, &installSignature)
+	if err != nil {
+		require.Nil(t, err)
+	}
+	assert.Equal(t, 3, len(installSignature))
+	installUUID := installSignature["install_id"]
+	_, err = uuid.Parse(installUUID)
+	assert.Nil(t, err)
```

Comment:
There is a dedicated method for this: `assert.NoError`. It has a better output upon failure.